src/app/[room]/page.tsx

```diff
@@ -1,31 +1,57 @@
 'use client'
 
-import { useEffect, useState } from 'react'
-import SocketIoInfo from '@/components/socketIoDevTools/SocketIoInfo'
+import { useEffect, useState, useRef } from 'react'
+// import SocketIoInfo from '@/components/socketIoDevTools/SocketIoInfo'
 import { socketRoomEmitter } from '@/services/socket'
 import RoomMainUi from '@/components/RoomMainUi'
-import { getRandomName } from '@/utils/sampleData'
+import RoomInfo from '@/components/RoomInfo'
+import NewUserDialog from '@/components/NewUserDialog'
 
-export default function UserRooms({ params }: { params: { room: string } }) {
+type Params = {
+	room: string
+}
+
+export default function UserRooms({ params }: { params: Params }) {
 	const [user, setUser] = useState('')
+	const [isJoinClicked, setIsJoinClicked] = useState(false)
+	const dialogRef = useRef<HTMLDialogElement>(null)
+	console.log('%c>>> params', 'color: red', params)
+
 	const { room } = params
 
 	useEffect(() => {
-		const user = getRandomName()
-		console.log('%c>>> roomId User params:', 'color: red', room, user)
-		setUser(user)
-		socketRoomEmitter('join-room', 'joined room', room, user)
-	}, [room])
+		if (dialogRef.current) {
+			dialogRef.current.showModal()
+		}
+	}, [])
+
+	useEffect(() => {
+		console.log('%c>>> user', 'color: red', user)
+		if (!user || !isJoinClicked) return
+		const timeStamp = Date.now().toString()
+		socketRoomEmitter('join-room', 'join', user, timeStamp, room)
```

Comment:
bug: Imagine, as a user, I refresh the page. I need to say my name again. A lot can go wrong here:
- If I say a different name, my old user is gone
- If I write the name of another user, then I can act on their behalf, even though it's a different person.

Did you realize this? How would you fix this? (It can be another MR, as it's a tricky bug)

Reply:
I was aware that re-entering the name was a pain. I hadn't thought about the other points. I'll save this one for another MR.